stellar-contract-env-host/src/test.rs

```diff
@@ -89,3 +89,31 @@ fn vec_as_seen_by_host() -> Result<(), ()> {
     assert_eq!(val0, val1);
     Ok(())
 }
+
+#[test]
+fn vec_front_and_back() -> Result<(), ()> {
+    let mut host = Host::default();
+    let scvec0: ScVec = ScVec(vec![ScVal::U32(1), ScVal::U32(2), ScVal::U32(3)].try_into()?);
+    let scobj0 = ScObject::Vec(scvec0);
+    let obj0 = host.to_host_obj(&scobj0).unwrap();
+    let front = unsafe {
+        <i32 as RawValConvertible>::unchecked_from_val(host.vec_front(*obj0.as_ref()).unwrap())
+    };
+    let back = unsafe {
+        <i32 as RawValConvertible>::unchecked_from_val(host.vec_back(*obj0.as_ref()).unwrap())
+    };
+    assert_eq!(front, 1);
```

Comment:
Instead of doing the unsafe unchecked you could compare the value returned with `ScVal::I32(1)`, etc. It might help keep the test focused on what it needs to test.

Reply:
Hmm not sure about this one. 
I think here comparing the raw value is what we want. `ScVal::I32` is what we used to construct the host object, the value gets unpacked from `ScVal`. Retrieving elements from the vector object gets back the `RawVal`, which doesn't know anything about `ScVal` any more.

Reply:
Ah yes. 👍🏻